### PR TITLE
Upd depts and insure compatibility

### PR DESCRIPTION
**Pull request by monatis**
- Updated depts. Workflows were broken after Pytorch 2.0 while installing the pinned 1.12.1.
- Added small compatibility code for pytorch-lightnin 1.9.0.
- Updated version to 0.1.35 to make a release with recently merged `CircleLoss` and `FastAPLoss`.

### CODE REVIEW

File: quaterion/train/callbacks/cleanup_callback.py [hunk 1]
```diff
@@ -18,7 +22,12 @@ def teardown(
             # If encoders were wrapped, unwrap them
             pl_module.unwrap_cache()
 
-            trainer.reset_train_val_dataloaders()
+            try:  # fix for pl>=1.9.0
+                trainer.reset_train_val_dataloaders()
+            except NotImplementedError:
+                trainer.reset_train_dataloader()
+                trainer.reset_test_dataloader
```
Comment (joein) [inline quaterion/train/callbacks/cleanup_callback.py:29]:
```suggestion
                trainer.reset_test_dataloader()
```

File: pyproject.toml [hunk 2]
```diff
@@ -31,7 +31,7 @@ ipdb = "^0.13.9"
 sphinx = ">=5.0.1"
 qdrant-sphinx-theme = { git = "https://github.com/qdrant/qdrant_sphinx_theme.git", branch = "master" }
 black = "^22.3.0"
-torchvision = "^0.12.0"
+torchvision = ">=0.12.0"
```
Comment (joein) [inline pyproject.toml:34]: why

Comment (monatis) [reply pyproject.toml:34]: It was fixing torchvision to 0.12.x (latest = 0.15.x) and the latest poetry couldn't verify its hash on CI, essentially braking our tests.

Comment (joein) [reply pyproject.toml:34]: yeah, my bad, I forgot that `^x.y.z` does not allow to increase `y` if `x == 0`

**Review by joein — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (monatis) [reply pyproject.toml:34]: Thanks!

**Pull request merged**